eslint with --rule option not working properly

### Steps to reproduce

Here is a example repo: https://github.com/tgreen7/lint-staged-eslint-rule

if you modify index.js, git add index.js, and run yarn precommit then no errors will be logged. However, if you uncomment the line `quotes: [2, 'double']` in the .eslintrc.js file and re-run the precommit then the error will be logged. 

Also you can try to run yarn lint, which adds that rule in the same way as the precommit, and it does get linted properly.

If I am passing that same rule to the precommit through the eslint --rule option then why is it not getting picked up while running?

### Debug Logs

<!--
Run `lint-staged` in debug mode using the command line flag `--debug`. You can
either modify your pre-commit script/command to include the flag or invoke
`lint-staged` manually(`./node_modules/.bin/lint-staged --debug`). Read more
about debug mode here -
https://github.com/okonet/lint-staged#command-line-flags.

NOTE: This requires `lint-staged` version >= `6.0.0`.
-->
<details>

<summary>expand to view</summary>

```
[taoh:.../lint-staged-test] ❯ ./node_modules/.bin/lint-staged --debug                                                   ✚ :: master  :: ⬡
  lint-staged:bin Running `lint-staged@7.0.0` +0ms
  lint-staged Loading config using `cosmiconfig` +0ms
  lint-staged Successfully loaded config from `/Users/taoh/Sites/lint-staged-test/package.json`:
  lint-staged { 'index.js': [ 'eslint --rule \'quotes: [2, double]\'' ] } +3ms
  lint-staged:cfg Normalizing config +0ms
  lint-staged:cfg Validating config +2ms
Running lint-staged with the following config:
{
  linters: {
    'index.js': [
      'eslint --rule \'quotes: [2, double]\''
    ]
  },
  concurrent: true,
  chunkSize: 9007199254740991,
  globOptions: {
    matchBase: true,
    dot: true
  },
  ignore: [],
  subTaskConcurrency: 1,
  renderer: 'verbose'
}
  lint-staged:run Running all linter scripts +0ms
  lint-staged:run Resolved git directory to be `/Users/taoh/Sites/lint-staged-test` +1ms
  lint-staged:run Loaded list of staged files in git:
  lint-staged:run [ 'index.js' ] +28ms
  lint-staged:gen-tasks Generating linter tasks +0ms
  lint-staged:cfg Normalizing config +35ms
  lint-staged:gen-tasks Generated task:
  lint-staged:gen-tasks { pattern: 'index.js',
  lint-staged:gen-tasks   commands: [ 'eslint --rule \'quotes: [2, double]\'' ],
  lint-staged:gen-tasks   fileList: [ '/Users/taoh/Sites/lint-staged-test/index.js' ] } +2ms
Running tasks for index.js [started]
  lint-staged:run-script Running script with commands [ 'eslint --rule \'quotes: [2, double]\'' ] +0ms
  lint-staged:cfg Normalizing config +5ms
eslint --rule 'quotes: [2, double]' [started]
  lint-staged:find-bin Resolving binary for command `eslint --rule 'quotes: [2, double]'` +0ms
  lint-staged:find-bin Binary for `eslint --rule 'quotes: [2, double]'` resolved to `/Users/taoh/Sites/lint-staged-test/node_modules/.bin/eslint` +3ms
  lint-staged:run-script bin: /Users/taoh/Sites/lint-staged-test/node_modules/.bin/eslint +5ms
  lint-staged:run-script args: [ '--rule',
  lint-staged:run-script   '\'quotes:',
  lint-staged:run-script   '[2,',
  lint-staged:run-script   'double]\'',
  lint-staged:run-script   '/Users/taoh/Sites/lint-staged-test/index.js' ] +0ms
  lint-staged:run-script opts: { reject: false } +1ms
eslint --rule 'quotes: [2, double]' [completed]
Running tasks for index.js [completed]
  lint-staged linters were executed successfully! +549ms
```

</details>

### Environment

<!-- Tell us about your development environment -->
- **OS:** macOS High Sierra 
- **Node.js:** v9.8.0
- **`lint-staged`:** v7.0.0

@tgreen7 This might be an eslint issue:

```
λ yarn lint
yarn run v1.5.1
$ eslint --rule 'quotes: [2, double]' index.js
Done in 1.28s.

λ .\node_modules\.bin\eslint.cmd --rule 'quotes: [2, double]' index.js
<no errors>

# Uncomment the rule in .eslint.js
λ .\node_modules\.bin\eslint.cmd index.js

E:\Projects\experiments\lint-staged-eslint-rule\index.js
  1:9  error  Strings must use doublequote  quotes

✖ 1 problem (1 error, 0 warnings)
  1 error, 0 warnings potentially fixable with the `--fix` option.
```

To be completely honest, this could also be hiding a `lint-staged` bug because I am not sure the way we split your command into arguments is completely correct when single quotes are present.

@sudo-suhas not sure what you mean. It works for me on this repo:

```
➜ eslint --rule 'quotes: [2, double]' index.js 

/Users/okonet/Projects/OSS/lint-staged/index.js
   3:1   error  Strings must use doublequote  quotes
   5:25  error  Strings must use doublequote  quotes
   6:26  error  Strings must use doublequote  quotes
   7:21  error  Strings must use doublequote  quotes
   9:24  error  Strings must use doublequote  quotes
  13:11  error  Strings must use doublequote  quotes
  13:34  error  Strings must use doublequote  quotes
  14:11  error  Strings must use doublequote  quotes
  14:26  error  Strings must use doublequote  quotes
  18:19  error  Strings must use doublequote  quotes
  21:7   error  Strings must use doublequote  quotes
  23:9   error  Strings must use doublequote  quotes

✖ 12 problems (12 errors, 0 warnings)
  12 errors, 0 warnings potentially fixable with the `--fix` option.
```

> not sure what you mean. It works for me on this repo:

It was a windows specific problem:
```
λ .\node_modules\.bin\eslint.cmd --rule 'quotes: [2, double]' index.js

λ .\node_modules\.bin\eslint.cmd --rule "quotes: [2, double]" index.js

E:\Projects\experiments\lint-staged-eslint-rule\index.js
  1:9  error  Strings must use doublequote  quotes

✖ 1 problem (1 error, 0 warnings)
  1 error, 0 warnings potentially fixable with the `--fix` option.
```

--- a/package.json
+++ b/package.json
@@ -27,7 +27,6 @@
   "dependencies": {
     "app-root-path": "^2.0.1",
     "chalk": "^2.3.1",
-    "cli-command-parser": "^1.0.3",
     "commander": "^2.14.1",
     "cosmiconfig": "^4.0.0",
     "debug": "^3.1.0",
@@ -46,6 +45,7 @@
     "pify": "^3.0.0",
     "please-upgrade-node": "^3.0.1",
     "staged-git-files": "1.1.1",
+    "string-argv": "^0.0.2",
     "stringify-object": "^3.2.2"
   },
   "devDependencies": {

--- a/src/findBin.js
+++ b/src/findBin.js
@@ -1,6 +1,6 @@
 'use strict'
 
-const parse = require('cli-command-parser')
+const parse = require('string-argv')
 const appRoot = require('app-root-path')
 const npmWhich = require('npm-which')(process.cwd())
 const checkPkgScripts = require('./checkPkgScripts')

--- a/test/findBin.spec.js
+++ b/test/findBin.spec.js
@@ -41,8 +41,17 @@ describe('findBin', () => {
   })
 
   it('should parse cmd and add arguments to args', () => {
-    const { bin, args } = findBin('my-linter task --fix --string "additional argument"')
+    const { bin, args } = findBin(
+      'my-linter task --fix --rule \'quotes: [2, double]\' --another "[complex:argument]"'
+    )
     expect(bin).toEqual('my-linter')
-    expect(args).toEqual(['task', '--fix', '--string', 'additional argument'])
+    expect(args).toEqual([
+      'task',
+      '--fix',
+      '--rule',
+      'quotes: [2, double]',
+      '--another',
+      '[complex:argument]'
+    ])
   })
 })

